ServerlessLibraryUI/src/App.vue

```diff
@@ -51,39 +51,61 @@
         updatedFilters: {}
       }
     },
-
+    props:{
+      language: String , 
+      type: String,
+      filtertext:String
+    },
     computed: {
-      list() {
-        //todo: this works but needs refactoring
-        var x = this.updatedFilters,
-          filter = new RegExp(x.filtertext, 'i'),
-          temp
-
-        temp = this.samples.filter(el =>
-          el.title.match(filter)
-          || el.description.match(filter)
-          || el.authortype.match(filter)
-          || el.repository.replace('https://github.com/','').match(filter)
-          || el.runtimeversion && el.runtimeversion.match(filter)
-        )
-        if (x.language && x.language.length > 0)
-          temp = temp.filter(el => el.language === String(x.language))
-        if (x.type && x.type.length > 0)
-          temp = temp.filter(el => el.type === String(x.type))
-        temp = temp.sort(
-          function (a, b) {
-            return b.totaldownloads - a.totaldownloads;
-          });
-        return temp
-      }
+      filters()
+      {
+         return {
+            language: this.language ? this.language:"",
+            type: this.type ? this.type : "",
+            filtertext: this.filtertext ? this.filtertext : ""
+        }
+      },
     },
 
     created() {
-      fetch('https://www.serverlesslibrary.net/api/Library')
+      if (!this.language && !this.filtertext && !this.type)
+      {
+        this.FetchSamples("");
+      }
+    },
+    methods: {
+      FetchSamples: function(queryString) {
+        fetch('https://serverlesslibrary.net/api/Library?' + queryString )
```

Comment:
don't you also need to make backend changes to support the query params?

Reply:
Added the changes for server side